crates/ika-core/src/dwallet_mpc/mpc_session.rs

```diff
@@ -397,7 +404,30 @@ impl DWalletMPCSession {
                     malicious_parties: vec![],
                 })
             }
-            MPCProtocolInitData::DecryptionKeyReshare(_) => todo!(),
+            MPCProtocolInitData::DecryptionKeyReshare(_) => {
+                let public_input = bcs::from_bytes(public_input)?;
+                let decryption_shares_primes = mpc_event_data
```

Comment:
Really confusing name. There is no prime here.. only in the encryption keys and proofs
```suggestion
                let decryption_key_shares = mpc_event_data
```